src/hitchhiker/tree/messaging.clj

```diff
@@ -198,37 +213,40 @@
   ([tree key]
    (lookup tree key nil))
   ([tree key not-found]
-   (let [path (core/lookup-path tree key)
-         expanded (apply-ops-in-path path)]
-     (get expanded key not-found))))
+   (go-try
+     (let [path (<? (core/lookup-path tree key))
+           expanded (apply-ops-in-path path)]
+       (get expanded key not-found)))))
 
 (defn insert
   [tree key value]
   (enqueue tree [(assoc (->InsertOp key value)
-                        :tag (java.util.UUID/randomUUID)
+                        :tag (uuid)
                         )]))
 
 (defn delete
   [tree key]
   (enqueue tree [(assoc (->DeleteOp key)
-                        :tag (java.util.UUID/randomUUID)
+                        :tag (uuid)
                         )]))
 
-(defn forward-iterator
-  "Takes the result of a search and returns an iterator going
+#?(:clj
+   (defn forward-iterator
```

Comment:
Ok, added https://github.com/datacrypt-project/hitchhiker-tree/issues/31 which should cover this.